config/software/git.rb

```diff
@@ -66,8 +66,19 @@
   patch source: "aix-use-freeware-install.patch", plevel: 1, env: patch_env if aix?
   patch source: "aix-strcmp-in-dirc.patch", plevel: 1, env: patch_env if aix?
 
+  if freebsd?
+    configure_args = ["--enable-pthreads=-pthread",
+                      "ac_cv_header_libcharset_h=no",
+                      "--with-curl=/usr/local",
+                      "--with-expat=/usr/local",
+                      "--with-perl=/usr/local/bin/perl"]
+  else
+    configure_args = []
+  end
+
   configure_command = ["./configure",
-                       "--prefix=#{install_dir}/embedded"]
+                       "--prefix=#{install_dir}/embedded",
+                       configure_args]
```

Comment:
This will create a nested Array. I think you want to declaure the shared conifgure_command Array first and then append it it:

```ruby
configure_command = ["./configure",
                     "--prefix=#{install_dir}/embedded",
                     configure_args]

if freebsd?
  configure_command << "--enable-pthreads=-pthread"
  configure_command << "ac_cv_header_libcharset_h=no"
  configure_command << "--with-curl=/usr/local"
  configure_command << "--with-expat=/usr/local"
  configure_command << "--with-perl=/usr/local/bin/perl"
end
```

That being said I don't think those paths are correct. You should be pointing at the curl, expat and perl that lives in `#{install_dir}/embedded`.